lumen/target.py

```diff
@@ -52,16 +51,19 @@ def __panel__(self):
     def _construct_layout(self):
         layout = self.layout
         if isinstance(layout, list):
-            if np.concatenate(layout).max() >= len(self.views):
-                raise ValueError(
-                    f"Layout for '{self.title}' can only have values up to {len(self.views) - 1}."
-                )
+            view_size = len(self.views)
             item = pn.Column(sizing_mode='stretch_both')
             for row_spec in layout:
                 row = pn.Row(sizing_mode='stretch_width')
                 for index in row_spec:
                     if isinstance(index, int):
-                        view = self.views[index]
+                        if index < view_size:
+                            view = self.views[index]
+                        else:
+                            raise ValueError(
+                                f"Layout for '{self.title}' can only "
+                                f"have values up to {view_size - 1}."
+                            )
```

Comment:
Being nit-picky about style here but I generally like writing `if <condition> then <error>` very explicitly for readability:

```suggestion
                        if index >= view_size:
                            raise ValueError(
                                f"Layout for '{self.title}' can only "
                                f"have values up to {view_size - 1}."
                            )
                        view = self.views[index]
```